src/vm/peimage.cpp

```diff
@@ -1258,7 +1258,7 @@ void PEImage::Load()
         {
             _ASSERTE(m_pLayouts[IMAGE_FLAT] != NULL);
 
-            if (!m_pLayouts[IMAGE_FLAT]->CheckILOnly())
+            if (!m_pLayouts[IMAGE_FLAT]->CheckILOnly() && !m_pLayouts[IMAGE_FLAT]->HasReadyToRunHeader())
```

Comment:
Would it better to modify `CheckILOnly` instead?

Reply:
(to make it consistent with `PEDecoder::IsILOnly`)

Reply:
I've seen this CheckILOnly get used in other places that seemed to care about IL only. Let me double check

Reply:
It seems possible to change CheckILOnly to return true for R2R images. I haven't really found anything in the code that would suggest otherwise. I'll make that change.